pygmt/figure.py

```diff
@@ -442,6 +463,8 @@ def show(self, dpi=300, width=500, method=None, waiting=0.5, **kwargs):
 
         # Set the display method
         if method is None:
+            if SHOW_CONFIG["method"] is None:
+                SHOW_CONFIG["method"] = _get_default_display_method()
```

Comment:
`SHOW_CONFIG["method"]` is set to the default display method when `Figure.show` is called the first time.